compiler/rustc_hir_typeck/src/fn_ctxt/inspect_obligations.rs

```diff
@@ -117,6 +118,17 @@ impl<'a, 'tcx> ProofTreeVisitor<'tcx> for NestedObligationsForSelfTy<'a, 'tcx> {
     }
 
     fn visit_goal(&mut self, inspect_goal: &InspectGoal<'_, 'tcx>) {
+        // No need to walk into goal subtrees that certainly hold, since they
+        // wouldn't then be stalled on an infer var.
+        if inspect_goal.result() == Ok(Certainty::Yes)
+            && !matches!(
```

Comment:
We currently report nested goals as `Certainty::Yes` even if they had nested goals... we could/should fix that?

Reply:
> normalizes-to goals?

Yes, typo.